README.md

```diff
@@ -1,3 +1,42 @@
 # medic-builds-repo
 
 Design doc for builds repository.
+
+DDOCS are [pushed up by travis](https://github.com/medic/medic-webapp/blob/master/scripts/ci/push_to_staging.sh) with a specific name. These are then [pulled back down by api](https://github.com/medic/medic-api/blob/master/controllers/upgrade.js) and deployed by [horticulturalist](https://github.com/medic/horticulturalist).
+
+## Naming scheme
+
+Three valid examples: `medic:medic:3.0.0-beta.1`, `medic:medic:3.0.0`, `medic:medic:my-awesome-feature-branch`.
+
+The first two chunks between `:` is the namespace and the application name respectively. The namespace is just that: it has no logic relevant and is just used to scope things. The application name maps to the original ddoc name. It is what will be used to determine what to call the ddoc when deploying it.
```

Comment:
I cannot stress enough that this is just a naming scheme. It may be hard to tell because everything else is hardcoded behind it, because I do not see the value in spending large amounts of time architecting and then building something we aren't sure about and don't currently need. My priority is to get this merged so we can start generating builds, and people can start testing branches using it, and we can start getting actual experience in this style of deployment. I have no doubt that large amounts of things will change between now and us being international deployment superstars. But, until we merge developers cannot branch off master and have their branches pushed to horti, and have people test their branches with horti, etc.

Anyway, on to the points as I read them:

The database it gets deployed to is not hard-coded into this scheme. It's not even mentioned in this readme. It is only hard-coded in the backend code because of the aforementioned YAGNI.

The ddoc name is mapped to the application name. However, if we ever feel like allowing this to not be true, we can just say that the ddoc is the application name *by default*, but you can either specify a different name in the metadata associated with the build[1], or pass a different one when you install it. Again, we don't support this because it asks a dozen questions of us we don't need to answer right now. I'm also not convinced of the merit of it (it doesn't have any value inside of medic afaict), so it's very low in the priority list, around when(/if) we support third parties deploying in this fashion.

It's definitely something we can consider when we are at that stage.

[1] We are currently using `build_time` from Kanso. When/If we drop Kanso or when/if we want other people to use this formally we can have our own metadata section.